subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/ui/SubscriptionsWebViewActivity.kt

```diff
@@ -284,11 +288,19 @@ class SubscriptionsWebViewActivity : DuckDuckGoActivity(), DownloadConfirmationD
     }
 
     private fun convertIntoSubscriptionWebViewActivityParams(intent: Intent): SubscriptionsWebViewActivityWithParams {
-        intent.getActivityParams(SubscriptionPurchase::class.java)?.let {
+        intent.getActivityParams(SubscriptionPurchase::class.java)?.let { subscriptionPurchaseActivityParams ->
             return SubscriptionsWebViewActivityWithParams(
                 url = BUY_URL,
-                origin = it.origin,
-            )
+                origin = subscriptionPurchaseActivityParams.origin,
+            ).let { webViewActivityWithParams ->
+                if (subscriptionPurchaseActivityParams.featurePage.isNullOrBlank().not()) {
+                    webViewActivityWithParams.copy(
+                        url = BUY_URL + "?$FEATURE_PAGE_QUERY_PARAM_KEY=${subscriptionPurchaseActivityParams.featurePage}",
```

Comment:
This is assuming that `BUY_URL` doesn't already have any query params, which is currently true for production, but not for staging environment. Also, someone might update `BUY_URL` in the future to include query params in production, and they will likely miss this. I think it would be safer to parse BUY_URL  to `Uri` and use `Uri.Builder` to append a query parameter to it.